### PR TITLE
Model card ++ for ABP, Phishing, Root cause analysis

### LINKED ISSUE #991: [DOC]: Update Model Cards to Model Card ++ format
### How would you describe the priority of this documentation request

High

### Please provide a link or source to the relevant docs

https://github.com/nv-morpheus/Morpheus/tree/branch-23.07/models#readme

### Describe the problems in the documentation

Model cards need to specifically address bias, explainability, privacy, safety and security per Nvidia's model card ++ format.

Model creators should update their own model cards.

- [x] ABP @gbatmaz 
- [ ] DFP @hsin-c 
- [ ] Fraud Detection @tzemicheal 
- [ ] Log parsing @raykallen 
- [x] Phishing @gbatmaz 
- [ ] Ransomware @raykallen 
- [x] Root cause @gbatmaz 
- [ ] SID @raykallen 

### (Optional) Propose a correction

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct
- [X] I have searched the [open documentation issues](https://github.com/nv-morpheus/Morpheus/issues?q=is%3Aopen+is%3Aissue+label%3Adoc) and have found no duplicates for this bug report

### PR DESCRIPTION
**Pull request by gbatmaz**
## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/nv-morpheus/Morpheus/blob/main/docs/source/developer_guide/contributing.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by mdemoret-nv** [comment]
@gbatmaz You are missing some copyright headers. Please run `python ci/scripts/copyright.py --git-modified-only --update-current-year --insert` and commit the changes to allow CI to pass.

File: models/model-cards/root-cause-analysis-model-card.md [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+* https://github.com/nv-morpheus/Morpheus/blob/branch-23.07/models/datasets/validation-data/root-cause-validation-data-input.jsonlines  <br>
+**Properties (Quantity, Dataset Descriptions, Sensor(s)):** 
+* kern.log files from DGX machines <br>
+**Dataset License:** 
+* N/A <br>
+
+## Inference:
+**Engine:** 
+* Triton <br>
+**Test Hardware:** <br>
```
Comment (hsin-c) [inline models/model-cards/root-cause-analysis-model-card.md:67]: Add new line above this line

File: models/model-cards/root-cause-analysis-model-card.md [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+* Not Applicable
+
+### Describe measures taken to mitigate against unwanted bias.
+
+* Not Applicable
+
+## Model Card ++ Explainability Subcard
+
+### Name example applications and use cases for this model. 
+* This model is an example of customized transformer-based root cause analysis. It can be further fine-tuned for specific root cause analysis or predictive maintenance needs and of your enterprise using the fine-tuning scripts in the repo. The hyper parameters can be optimised to adjust to get the best results with your dataset. The aim is to get the model to predict some false positives that could be previously unknown error types. Users can use this root cause analysis method with other log types too. If they have known failures in their logs, they can use them to train along with ordinary logs and can detect other root causes they weren't aware of before.
```
Comment (hsin-c) [inline models/model-cards/root-cause-analysis-model-card.md:125]: If the pretrained model is, in addition to being fine-tunable, used for testing, let's add the mention of that too.

File: models/model-cards/root-cause-analysis-model-card.md [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+
+### Fill in the blank for the model technique.
+
+* This model is intended for developers that want to build and/or customize root cause analysis models.
```
Comment (hsin-c) [inline models/model-cards/root-cause-analysis-model-card.md:129]: If the pretrained model is not readily useable for developers (they need to train from scratch with their own data), let change this to say it's for testing only (similar to the two cards above)

File: models/model-cards/root-cause-analysis-model-card.md [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+
+### Name who is intended to benefit from this model. 
+
+* This model is intended for users who use language models for root cause analysis.
```
Comment (hsin-c) [inline models/model-cards/root-cause-analysis-model-card.md:133]: We can either update this or add another bullet point to mention the testing usage

File: models/model-cards/root-cause-analysis-model-card.md [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+
+### Describe physical safety impact (if present).
+* N/A
+
+### Was model and dataset assessed for vulnerability for potential form of attack?
+* No
+
+### Name applications for the model.
+
+* Root Cause Analysis, enabling predictive maintenance.
```
Comment (hsin-c) [inline models/model-cards/root-cause-analysis-model-card.md:176]: Similarly, replace to add a new bullet point for testing purposes.
`The primary application for this model is testing the Morpheus pipeline.`

File: models/model-cards/root-cause-analysis-model-card.md [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+* N/A
+
+### Is data in dataset traceable?
+* N/A
+
+### Scanned for malware?
+* N/A
+
+### Are we able to identify and trace source of dataset?
+* N/A
```
Comment (hsin-c) [inline models/model-cards/root-cause-analysis-model-card.md:231]: Did we make this dataset? If so, it should be traceable (as we're the maker of it) and we can say `Yes` here.

**Review by hsin-c — CHANGES_REQUESTED** [review]

File: models/model-cards/phishing-model-card.md [hunk 7]
```diff
@@ -0,0 +1,240 @@
...
+
+## Model Architecture: 
+**Architecture Type:** 
+* Transformers <br>
+**Network Architecture:** 
+* BERT <br>
+
+## Input: (Enter "None" As Needed)
+**Input Format:** 
+* smsspamcollection.zip <br>
```
[the remark this answers is not in the document]
Comment (gbatmaz) [reply models/model-cards/phishing-model-card.md:38]: yes the validation script uses zip file and then tabular data gets extracted. I will explain what is meant in more detail.

File: models/model-cards/root-cause-analysis-model-card.md (hunk 6 above)
Comment (gbatmaz) [reply models/model-cards/root-cause-analysis-model-card.md:249]: We didn't create the content dataset. Will give more details offline.

File: models/model-cards/abp-model-card.md [hunk 8]
```diff
@@ -0,0 +1,239 @@
...
+
+## Description:
+* This model is an example of a binary XGBoost classifier to differentiate between anomalous GPU behavior, such as crypto mining / GPU malware, and non-anomalous GPU-based workflows (e.g., ML/DL training). The model is an XGBoost model. <br>
+
+## References(s):
+* Chen, Guestrin (2016) XGBoost. A scalable tree boosting system. https://arxiv.org/abs/1603.02754  <br> 
+
+## Model Architecture: 
+**Architecture Type:** 
+* Gradient boosting <br>
```
[the remark this answers is not in the document]
Comment (gbatmaz) [reply models/model-cards/abp-model-card.md:35]: added

File: models/model-cards/abp-model-card.md [hunk 9]
```diff
@@ -0,0 +1,239 @@
...
+**Network Architecture:** 
+* XGBOOST <br>
+
+## Input: (Enter "None" As Needed)
+**Input Format:** 
+* nvidia-smi output <br>
+**Input Parameters:**
```
[the remark this answers is not in the document]
Comment (gbatmaz) [reply models/model-cards/abp-model-card.md:47]: added

File: models/model-cards/root-cause-analysis-model-card.md (hunk 2 above)
Comment (gbatmaz) [reply models/model-cards/root-cause-analysis-model-card.md]: added

**Review by hsin-c — APPROVED** [review]

**Pull request merged**